### PR TITLE
chore(): remove use of "bind" proposal code

### PR DESCRIPTION
**Pull request by boneskull**
- Removes the experimental "function bind" syntax code
- Removes a bunch of babel plugins; replaces with "shipped proposals" setting in babel-env
- build now targets Node.js 12
- upgrade eslint; remove now-unnecessary babel eslint parser

The output of the build before and after this change looks identical.

See #136

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -4,17 +4,12 @@ module.exports = {
       '@babel/preset-env',
       {
         targets: {
-          node: 10,
+          node: 12,
```
Comment (wswebcreation) [inline babel.config.js:7]: Should we go for 14 because of EOL of 12 is around now

**Review by wswebcreation — CHANGES_REQUESTED** [review]
LGTM, 1 small remark

File: babel.config.js (hunk 1 above)
Comment (boneskull) [reply babel.config.js:7]: yes, but that's a breaking change.  kind of depends how much this will negatively impact customers

Comment (wswebcreation) [reply babel.config.js:7]: Ok, let’s merge it then

**Review by wswebcreation — APPROVED** [review]

**Pull request merged**